### PR TITLE
[NUI] Use weak reference for theme changed event

### PR DESCRIPTION
**Pull request by rabbitfor**
Signed-off-by: Jiyun Yang <ji.yang@samsung.com>

### Description of Change ###
Previous theme changed event was holding reference of the view for the theme manager's lifetime
unless the view is disposed by calling Dispose() explicitly.
Since the theme manager is a static class, so there might be a change that the view can't get free by GC.
To solve this problem, this patch apply a weak event that uses weak reference for the handler
so that the views are no longer be caught by theme manager.

### API Changes ###

 - ACR:

### CODE REVIEW

File: src/Tizen.NUI/src/internal/WeakEvent.cs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+                {
+                    item.Invoke(sender, args);
+                    continue;
+                }
+                handlers.Remove(item);
+            }
+        }
+
+        internal class WeakHandler<T>
+        {
```
Comment (huiyueun) [inline src/Tizen.NUI/src/internal/WeakEvent.cs:53]: I think warning of new file should be removed.

**Comment by huiyueun** [comment]
Please add detailed comment about this patch.

File: src/Tizen.NUI/src/internal/WeakEvent.cs (hunk 1 above)
Comment (rabbitfor) [reply src/Tizen.NUI/src/internal/WeakEvent.cs:53]: Done :)

**Comment by rabbitfor** [comment]
> Please add detailed comment about this patch.
Done :)

**Review by dongsug-song — APPROVED** [review]
(b^_^)b
I think this would be very good if this is applied previous existing events.
do you think this is possible?
almost all events are connected to native dali callbacks and I am not sure this Weak-Reference-Event is feasible for native binding events.
could you figure this out?
thank you.

**Comment by dongsug-song** [comment]
@rabbitfor 
Hello, could you check my comment above? ^^;
thank you.

**Pull request merged**